src/blockchain/calls/calls.ts

```diff
@@ -62,10 +55,30 @@ function calls([context, account]: [NetworkConfig, string]) {
   };
 }
 
+function readCalls(context: NetworkConfig) {
+  const call = callCurried(context);
+
+  return {
+    otcGetBuyAmount: call(getBuyAmount),
+    otcGetPayAmount: call(getPayAmount),
+    otcGetOffersAmount: call(getOffersAmount),
+    otcGetBestOffer: call(getBestOffer),
+    otcOffers: call(offers),
+  };
+}
+
 export const calls$ = combineLatest(context$, initializedAccount$).pipe(
   map(calls),
 );
 
+export const readCalls$ = context$.pipe(
+  map(readCalls),
+);
+
 export type Calls$ = typeof calls$;
 
 export type Calls = ObservableItem<Calls$>;
+
+export type ReadCalls$ = typeof readCalls$;
```

Comment:
I would do it differently.

I will have types `Readable | Writable`.
So everywhere where `calls$` are used , I will have a single argument of this type. `Writable` will extends the readable methods with write ones. On higher level you will decide whether to build `calls$` read only or with write access as well. but the passed pipeline will only one

Reply:
This is also what Maciek suggested, we can talk about it.